app/cdap/components/shared/ConfigurationGroup/index.tsx

```diff
@@ -154,7 +164,8 @@ const ConfigurationGroupView: React.FC<IConfigurationGroupProps> = ({
     if (params.updateFilteredConfigurationGroups) {
       fcg = filterByCondition(configurationGroups, widgetJson, pluginProperties, changedValues);
     }
-
+    // update currentValues state with whatever changed
+    replaceDifferenceInObjects(currentValues, changedValues);
```

Comment:
Referring to [this](https://github.com/cdapio/cdap-ui/pull/467/files#r860314667), should we simply use `setCurrentValues(changedValues)`?

Reply:
`currentValues` have all the keys, `changedValues` may contain partial keys since some will get deleted in previous `removeFilteredProperties` operation. Also it's async so I'm worried when I want to use the `currentValues` later, it's not getting the updated value

Reply:
Give this a try and see if it works. We probably don't want to mutate our state object.
```javascript
setCurrentValues(changedValues)
useEffect(() => {
    const updatedFilteredValues = removeFilteredProperties(changedValues, fcg);
    changeParentHandler(updatedFilteredValues);
  }, [currentValues]);
```

Reply:
turns out using setCurrentValues is fine